access/slackbot/main.go

```diff
@@ -263,49 +293,63 @@ func (a *App) Run(ctx context.Context) error {
 	// Create callback server prividing a.OnSlackCallback as a callback function
 	a.callbackServer = NewCallbackServer(&a.conf, a.OnSlackCallback)
 
+	// Create separate context for Watcher in order to terminate it individually.
+	wctx, wcancel := context.WithCancel(ctx)
+
+	var jobs sync.WaitGroup
+	a.spawn = func(f func(context.Context)) {
+		jobs.Add(1)
+		go func() {
+			f(ctx)
+			jobs.Done()
+		}()
+	}
+
+	jobs.Add(1) // Start the "app" job. We have to do it for Wait() not being returned beforehand.
+	a.doneCh = make(chan struct{})
+	doneCh := a.doneCh // Save the variable to access it when unlocked
+	go func() {
+		jobs.Wait()
+		close(doneCh)
+	}()
```

Comment:
It looks like we don't even need this goroutine. You can just defer this channel closure and use `jobs.Wait()` in the end

Reply:
👍